js/actions/tests/loadSourceText.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
+
+const { createStore } = require("../../util/test-head");
+const { loadSourceText } = require("../../actions");
+const { getSourceText } = require("../../queries");
+
+var expect = require("chai").expect;
+
+const sourceText = {
+  "foo1": { source: "function() {\n  return 5;\n}",
+            contentType: "text/javascript" },
+  "foo2": { source: "function(x, y) {\n  return x + y;\n}",
+            contentType: "text/javascript" }
+};
+
+const mockThreadClient = {
+  source: function(form) {
+    return {
+      source: () => {
+        return new Promise((resolve, reject) => {
+          resolve(sourceText[form.actor]);
+        });
+      }
+    };
+  }
+};
+
+describe("loadSourceText", () => {
```

Comment:
I split two source actions loadSourceText and newSource to their own test files... I'm not sure if that's necessary. Maybe here, but not with other actions... I guess we'll figure that out.